streams/src/test/java/org/apache/kafka/streams/state/internals/StoreQueryUtilsTest.java

```diff
@@ -57,6 +60,30 @@ public void shouldReturnErrorOnNullContext() {
         );
     }
 
+    @Test
+    public void shouldReturnErrorOnNullContextForVersionedKeyQueries() {
```

Comment:
Wondering if we need this test? The logic that triggers this "not up to bound" error is generic and applies to all query types, and is not specific to the newly added query. Seems we just add a redundant test case?